utils/cq/daos_pylint.py

```diff
@@ -0,0 +1,404 @@
+#!/usr/bin/env python3
+"""Wrapper script for calling pylint"""
+
+import os
+import re
+from collections import Counter
+import subprocess  # nosec
+import argparse
+from pylint.lint import Run
+from pylint.reporters.collecting_reporter import CollectingReporter
+from pylint.lint import pylinter
+
+# Pylint checking for the DAOS project.
+
+# Wrapper script for pylint that does the following:
+#  Wraps scons commands and adjusts report line numbers
+#  Sets python-path or similar as required
+#  Runs in parallel across whole tree
+#  Supports minimum python version
+#  Supports venv usage
+#  Can be used by atom.io live
+#  Outputs directly to github annotations
+# To be added:
+#  Can be used in Jenkins to report regressions
+#  Can be used as a commit-hook
+#  flake8 style --diff option
+
+# For now this spilts code into one of three types, build (scons), ftest or other.  For build code
+# it enforces all style warnings except f-strings, for ftest it sets PYTHONPATH correctly and
+# does not warn about f-strings, for others it runs without any special flags.
+
+# Errors are reported as annotations to PRs and will fail the build, as do warnings in the build
+# code.  The next step is to enable warnings elsewhere to be logged, but due to the large number
+# that currently exist in the codebase we need to restrict this to modified code.  Spellings can
+# also be enabled shortly however we have a number to correct or whitelist before enabling.
+
+
+class WrapScript():
+    """Create a wrapper for a scons file and maintain a line mapping"""
+
+    def __init__(self, fname):
+
+        # TODO: Use a NamedTemporaryFile file here, it's an easy change to make but there appears
```

Comment:
(lint) Use a NamedTemporaryFile file here, it's an easy change to make but there appears